pkg/runtime/scheme.go

```diff
@@ -153,6 +153,38 @@ func (s *Scheme) AddUnversionedTypes(version unversioned.GroupVersion, types ...
 	}
 }
 
+func validateListType(t reflect.Type) {
+	// exceptions
+	if t.Name() == "APIGroupList" || t.Name() == "APIResourceList" {
```

Comment:
This is so wrong I don't even know where to start.  If we have to do this here, then we can't correctly solve this at all (because we're already in violation).